vllm/v1/core/sched/scheduler.py

```diff
@@ -656,10 +664,16 @@ def update_from_output(
                 # Invariant: EngineCore returns no partial prefill outputs.
                 assert not prompt_logprobs_tensors
 
-            self.scheduled_req_ids.remove(req_id)
+            self.scheduled_req_ids[req_id] -= 1
+            if self.scheduled_req_ids[req_id] == 0:
+                del self.scheduled_req_ids[req_id]
```

Comment:
Why do we delete? IIUC, this is a redundant operation. If the request is not finished, the request will be added again to `scheduled_req_ids`.

Reply:
Ditto